tests/instruments/test_instrument.py

```diff
@@ -23,13 +23,52 @@
 #
 
 
+from unittest import mock
+
 import pytest
 
 from pymeasure.instruments import Instrument
+from pymeasure.instruments.instrument import DynamicProperty
+from pymeasure.adapters import FakeAdapter
+from pymeasure.test import expected_protocol
 from pymeasure.instruments.fakes import FakeInstrument
 from pymeasure.instruments.validators import strict_discrete_set, strict_range, truncated_range
 
 
+class GenericInstrument(FakeInstrument):
+    #  Use truncated_range as this easily lets us test for the range boundaries
+    fake_ctrl = Instrument.control(
+        "", "%d", "docs",
+        validator=truncated_range,
+        values=(1, 10),
+        dynamic=True,
+    )
+    fake_setting = Instrument.setting(
+        "%d", "docs",
+        validator=truncated_range,
+        values=(1, 10),
+        dynamic=True,
+    )
+    fake_measurement = Instrument.measurement(
+        "", "docs",
+        values={'X': 1, 'Y': 2, 'Z': 3},
+        map_values=True,
+        dynamic=True,
+    )
+
+
+@pytest.fixture(scope="module")
+def generic():
+    return GenericInstrument()
+
+
+class NewRangeInstrument(GenericInstrument):
+    # Choose different properties' values, like you would for another device model
+    fake_ctrl_values = (10, 20)
+    fake_setting_values = (10, 20)
+    fake_measurement_values = {'X': 4, 'Y': 5, 'Z': 6}
+
```

Comment:
Can you also move ExtendedInstrument and StrictExtendedInstrument up here, so that all the related classes are in one place?

Reply:
At first I did it as you suggested, but then I moved it back down to their usage...

Reply:
hm, ok, that's also a fair point, and only taste in the end.